commands/run.go

```diff
@@ -0,0 +1,308 @@
+				RestartOnFailure:    true,
+				RestartOnCompletion: true,
+			})
+		}
+
+		// Add several state tasks to read actor state from each indexed block
+		for i := 0; i < cctx.Int("actorstate-workers"); i++ {
+			p, err := actorstate.NewActorStateProcessor(rctx.db, rctx.api, cctx.Duration("actorstate-lease"), cctx.Int("actorstate-batch"), heightFrom, heightTo, actorCodes)
+			if err != nil {
+				return err
+			}
+			scheduler.Add(schedule.TaskConfig{
+				Name:                fmt.Sprintf("ActorStateProcessor%03d", i),
+				Task:                p,
+				RestartOnFailure:    true,
+				RestartOnCompletion: true,
+			})
+		}
+
+		// Add several message tasks to read messages from indexed tipsets
+		for i := 0; i < cctx.Int("message-workers"); i++ {
+			scheduler.Add(schedule.TaskConfig{
+				Name:                fmt.Sprintf("MessageProcessor%03d", i),
+				Task:                message.NewMessageProcessor(rctx.db, rctx.api, cctx.Duration("message-lease"), cctx.Int("message-batch"), heightFrom, heightTo),
+				RestartOnFailure:    true,
+				RestartOnCompletion: true,
+			})
+		}
+
+		// Start the scheduler and wait for it to complete or to be cancelled.
+		err = scheduler.Run(ctx)
+		if !errors.Is(err, context.Canceled) {
+			return err
+		}
+		return nil
+	},
+}
+
+// getActorCodes parses the cli flags to obtain a list of actor codes for the actor state processor. We support some
+// common short names for actors or the cid of the actor code.
+func getActorCodes(cctx *cli.Context) ([]cid.Cid, error) {
+	include := cctx.StringSlice("actorstate-include")
+	exclude := cctx.StringSlice("actorstate-exclude")
+	if len(include) == 0 && len(exclude) == 0 {
+		// By default we process all supported actor types
+		return actorstate.SupportedActorCodes(), nil
+	}
+
+	if len(include) != 0 && len(exclude) != 0 {
+		return nil, fmt.Errorf("cannot specify both actorstate-include and actorstate-exclude")
+	}
+
+	if len(include) != 0 {
+		return parseActorCodes(include)
+	}
+
+	excludeCids, err := parseActorCodes(include)
+	if err != nil {
+		return nil, err
+	}
+
+	var codes []cid.Cid
+	for _, c := range actorstate.SupportedActorCodes() {
+		excluded := false
+		for _, ex := range excludeCids {
+			if c.Equals(ex) {
+				// exclude it
+				excluded = true
+				break
+			}
+		}
+		if !excluded {
+			codes = append(codes, c)
+		}
+	}
+
+	if len(codes) == 0 {
+		return nil, fmt.Errorf("all supported actors have been excluded")
+	}
+
+	return codes, nil
+}
+
+func parseActorCodes(ss []string) ([]cid.Cid, error) {
+	var codes []cid.Cid
+	for _, s := range ss {
+		c, ok := actorNamesToCodes[s]
+		if ok {
+			codes = append(codes, c)
+			continue
+		}
+
+		var err error
+		c, err = cid.Decode(s)
+		if err != nil {
+			return nil, fmt.Errorf("invalid cid: %w", err)
+		}
+		codes = append(codes, c)
+	}
+
+	return codes, nil
+}
+
+var actorNamesToCodes = map[string]cid.Cid{
+	"fil/2/system":           builtin.SystemActorCodeID,
```

Comment:
can you use [`ActorNameByCode`](https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/codes.go#L77) to get these, rather than duplicating them?

Reply:
I think we're going to remove these literals in exchange for tokens which are easy to pass into the CLI... (ex: `storageminer` -> all versions of `/fil/X/storageminer`)